bin/github-backup

```diff
@@ -382,24 +389,67 @@ def get_github_repo_url(args, repository):
     return repo_url
 
 
+def _update_githubapp_token(current_token):
+    global _token_cache
+    token_path = os.path.expanduser('~/.gh-token')
+
+    if os.path.exists(token_path):
+        last_modified = os.path.getmtime(token_path)
+
+        if last_modified > _token_cache['last_modified']:
+            with open(token_path, "r") as f:
+                new_token = f.read().strip()
+            _token_cache['token'] = new_token
+            _token_cache['last_modified'] = last_modified
+
+        if current_token != _token_cache['token']:
+            return _token_cache['token']
+    else:
+        log_error('No token found in ~/.gh-token')
+
+    return current_token
+
+
 def retrieve_data(args, template, query_args=None, single_request=False):
     auth = get_auth(args)
     query_args = get_query_args(query_args)
     per_page = 100
     page = 0
     data = []
 
+    if args.token:
+        auth = _update_githubapp_token(auth)
```

Comment:
I think you should move the `_update_githubapp_token()` and `add_header()` calls into `_construct_request()`. We always need an up to date token and headers in the request. No need to let this lower level detail pollute the higher level usages